### PR TITLE
hw:6.1-pair-generric solved

### PR DESCRIPTION
**Pull request by Danhov**

### CODE REVIEW

File: src/main/java/core/basesyntax/Pair.java [hunk 1]
```diff
@@ -1,5 +1,47 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<T1, T2> {
+    T1 first;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Pair.java]: bad naming. T and K suggested or something like it. and need access modifiers, please

File: src/main/java/core/basesyntax/Pair.java [hunk 2]
```diff
@@ -1,5 +1,47 @@
...
+    T2 second;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Pair.java]: same

File: src/main/java/core/basesyntax/Pair.java [hunk 3]
```diff
@@ -1,5 +1,47 @@
...
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        Pair<T1, T2> pair = (Pair<T1, T2>) o;
+
+        if (first != null ? !first.equals(pair.first) : pair.first != null) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Pair.java]: looks weird. try to simplify

File: src/main/java/core/basesyntax/Pair.java [hunk 4]
```diff
@@ -1,5 +1,47 @@
...
+            return false;
+        }
+        return second != null ? second.equals(pair.second) : pair.second == null;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Pair.java]: same about this return. try to compare two fields in a single expression

File: src/main/java/core/basesyntax/Pair.java [hunk 5]
```diff
@@ -1,5 +1,49 @@
 package core.basesyntax;
 
-class Pair {
+import java.util.Objects;
 
+class Pair<T, K> {
+    T first;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Pair.java]: please, add access modifier. I wrote you that at previous review, be careful

File: src/main/java/core/basesyntax/Pair.java [hunk 6]
```diff
@@ -1,5 +1,49 @@
...
+    K second;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Pair.java]: same

File: src/main/java/core/basesyntax/Pair.java (hunk 5 above)
Comment (Danhov) [reply src/main/java/core/basesyntax/Pair.java]: Sorry about that, my mistake

**Review by Tarabarnik — APPROVED** [review]

**Pull request closed without merging**